src/InsertionsClient.Core/Api/Providers/InsertionApi.cs

```diff
@@ -225,10 +227,21 @@ internal bool TryExtractManifestAssets(string manifestFile, ICollection<Asset> a
 
                 Trace.WriteLine($"De-serialized {buildManifest.Builds.Count} builds from the manifest file.");
 
-                int initialCollectionSize = assets.Count;
+                if (!buildManifest.Builds.Any(b => b?.Assets != null && b.Assets.Any()))
```

Comment:
minor perf:  is `Any()` cheaper than `Count()` ?  And if the data structure is an array, than `Length`?

Reply:
It appears to be slightly worse in performance, but it helps us ignore the underlying type here so I chose to go with this.